migrate/pkg/migrate/source/github.go

```diff
@@ -86,70 +44,40 @@ func (g *Github) parse(sourceURL string) (*config, error) {
 	c.Repo = p[0]
 	c.Path = strings.Join(p[1:], "/")
 	c.Ref = u.Fragment
-	c.Username = u.User.Username()
-	c.Password, _ = u.User.Password()
 
 	return c, nil
 }
 
-func (g *Github) getSourceCodeDirPath(tmpDir string, c *config) string {
-	return filepath.Join(
-		tmpDir,
-		fmt.Sprintf("%s-%s-%s", c.Owner, c.Repo, c.Ref),
+func (g *Github) downloadFromClone(c *config) (string, error) {
+	sourceCodeDirPath := g.getSourceCodeDirPath(g.CacheDir, c, "clone")
+	cloneSource := fmt.Sprintf(
+		"https://github.com/%s/%s.git",
+		c.Owner,
+		c.Repo,
 	)
-}
-
-// getSrcFolderName searches and return the folder name inside the destination dir
-// destination dir should only contain one folder
-func getSrcFolderName(destinationDir string) (string, error) {
-	var paths []string
-	files, err := ioutil.ReadDir(destinationDir)
-	if err != nil {
-		return "", fmt.Errorf("unable to find source folder: %v", err.Error())
-	}
 
-	for _, file := range files {
-		if file.IsDir() {
-			paths = append(paths, file.Name())
+	if _, e := os.Stat(sourceCodeDirPath); os.IsNotExist(e) {
+		cmd := exec.Command("git", "clone", cloneSource, sourceCodeDirPath)
+		err := cmd.Run()
```

Comment:
Connect the process stdout and stderr to those of this process. So we can see what git is doing.

```go
cmd := exec.Command(...)
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
err := cmd.Run()
```

Reply:
Since we used logrus in the migrate program, so I pipe to logrus instead. See if it looks good?

Reply:
Great!